airbyte-integrations/connectors/source-mongodb-v2/src/test/java/io/airbyte/integrations/source/mongodb/MongoDbReplicaSetTest.java

```diff
@@ -0,0 +1,283 @@
+      final MongoCollection<Document> movieCollection = client.getDatabase(DB_NAME).getCollection(COLLECTION_NAME);
+
+      // Insert a new document
+      movieCollection.insertOne(new Document().append("_id", new ObjectId())
+          .append("title", insertedTitle).append("catalogId", DATASET_SIZE + 1));
+
+      // Update an existing document
+      final Document updateQuery = new Document();
+      updateQuery.append("catalogId", new Document("$eq", 1234));
+      movieCollection.updateOne(updateQuery, Updates.set("title", updatedTitle));
+
+      // Delete an existing document
+      final Bson deleteQuery = Filters.eq("catalogId", 999);
+      final Document deletedDocument = movieCollection.findOneAndDelete(deleteQuery);
+
+      final Document filter = new Document();
+      filter.put("ns", DB_NAME + "." + COLLECTION_NAME);
+      filter.put("op", new Document("$in", Arrays.asList("i", "u", "d")));
+      filter.put("ts", new Document().append("$gte", new BsonTimestamp(now, 1)));
+      final Document projection = new Document("ts", 1).append("op", 1).append("o", 1);
+      final Document sort = new Document("$natural", 1);
+
+      final MongoCursor<Document> cursor = oplog
+          .find(filter)
+          .projection(projection)
+          .sort(sort)
+          .cursorType(CursorType.TailableAwait)
+          .noCursorTimeout(true)
+          .cursor();
+
+      final Collection<Document> changes = new ArrayList<>();
+
+      while (true) {
+        final Document document = cursor.tryNext();
+        if (document == null) {
+          break;
+        } else {
+          LOGGER.info("{}", document);
+          changes.add(document);
+        }
+      }
+
+      assertEquals(3, changes.size());
+
+      assertTrue(changes.stream().filter(d -> "i".equals(d.get("op"))).findFirst().isPresent());
+      final Document insertedDocument = (Document) changes.stream().filter(d -> "i".equals(d.get("op"))).findFirst().get().get("o");
+      assertEquals(insertedTitle, insertedDocument.get("title"));
+
+      assertTrue(changes.stream().filter(d -> "u".equals(d.get("op"))).findFirst().isPresent());
+      final Document updatedDocument = (Document) changes.stream().filter(d -> "u".equals(d.get("op"))).findFirst().get().get("o");
+      final Document updatedDiff = ((Document) ((Document) updatedDocument.get("diff")).get("u"));
+      assertEquals(updatedTitle, updatedDiff.get("title"));
+
+      assertTrue(changes.stream().filter(d -> "d".equals(d.get("op"))).findFirst().isPresent());
+      final Document deletedDocumentOpLog = (Document) changes.stream().filter(d -> "d".equals(d.get("op"))).findFirst().get().get("o");
+      assertEquals(deletedDocument.get("_id"), deletedDocumentOpLog.get("_id"));
+    }
+  }
+
+  private static String buildReplicaSetConfig() {
+    return "{_id:\\\"" + REPLICA_SET_ID + "\\\"," +
+        "members:[{_id:0,host:\\\"" + MONGO_DB1_NAME + "\\\"}," +
+        "{_id:1,host:\\\"" + MONGO_DB2_NAME + "\\\"}," +
```

Comment:
could be updated to use java text blocks with the [formatted](https://docs.oracle.com/en/java/javase/17/text-blocks/index.html#string-formattedobject...-args) method

Reply:
@colesnodgrass I believe I did that and IntelliJ changed it back to string concatenation, but I will try again

Reply:
That wouldn't surprise me, I didn't even know the `formatted` method existed until reviewing this PR.  That might be for a reason, maybe it doesn't perform well.

Reply:
This has been addressed.